### PR TITLE
376991: Add BCP search endpoint

### PR DESCRIPTION
**Pull request by reece-bennett**
[AB#376991](https://dev.azure.com/defragovuk/54a51ea2-a7e4-4951-98ac-e063e18ccb63/_workitems/edit/376991)

### CODE REVIEW

File: src/STW.StubApi.Presentation/Services/Bcp/BcpController.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+[ApiController]
+[Route("bcps")]
+public class BcpController
+{
+    private static readonly List<BcpSearchResultDto.BcpDto> Bcps =
+    [
+        new BcpSearchResultDto.BcpDto
+        {
+            Id = 1,
```
Comment (sean-treanor) [inline src/STW.StubApi.Presentation/Services/Bcp/BcpController.cs:14]: These might be better placed in another file so we don't clutter the controller.

File: src/STW.StubApi.Presentation/Services/Bcp/BcpController.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ];
+
+    [HttpGet("search")]
+    [Consumes("application/json")]
+    public BcpSearchResultDto Search([FromQuery] string code)
+    {
+        return new BcpSearchResultDto
+        {
+            Page = 1,
+            Elements = 25,
```
Comment (sean-treanor) [inline src/STW.StubApi.Presentation/Services/Bcp/BcpController.cs:34]: Does this represent the max items we want per page or the number of elements returned?

Comment (reece-bennett) [reply src/STW.StubApi.Presentation/Services/Bcp/BcpController.cs:34]: From IPAFFS it looks like max items per page - confusing I know :D

File: src/STW.StubApi.Presentation/Services/Bcp/BcpController.cs (hunk 1 above)
Comment (reece-bennett) [reply src/STW.StubApi.Presentation/Services/Bcp/BcpController.cs:14]: I don't mind them here, if there were more of them I'd agree

**Review by kamilmkainos — APPROVED** [review]

**Review by sean-treanor — APPROVED** [review]

**Pull request merged**